rules-reviewed/eap7/eap6/hsearch.windup.xml

```diff
@@ -0,0 +1,880 @@
+            <perform>
+                <hint title="Deprecated annotation Key in Hibernate Search 5" effort="1" severity="optional">
+                    <message>Custom filter cache keys are a deprecated feature and are scheduled to be removed in Hibernate Search 6. As of Hibernate Search 5.1, the filter cache keys are determined automatically based on the filter parameters so it is no longer required to provide a key object. </message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscelanous Changes in Hibernate Search 5.x" />
+                    <link href="https://docs.jboss.org/hibernate/search/5.5/api/index.html?org/hibernate/search/annotations/Key.html"
+                        title="Hibernate Search javadoc for @Key" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00119">
+            <when>
+                <javaclass references="org.hibernate.search.annotations.CacheFromIndex">
+                    <location>ANNOTATION</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Deprecated annotation CacheFromIndex in Hibernate Search 5" effort="1" severity="optional">
+                    <message> Remove the annotation. No alternative replacement is necessary.</message>
+                    <link
+                        href="https://access.qa.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/version-7.0/migration-guide/#migrate_miscellaneous_hibernate_search_changes"
+                        title="Miscelanous Changes in Hibernate Search 5.x" />
+                    <link
+                        href="https://docs.jboss.org/hibernate/search/5.5/api/index.html?org/hibernate/search/annotations/CacheFromIndex.html"
+                        title="Hibernate Search javadoc for @CacheFromIndex" />
+                    <tag>hibernate-search</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="hsearch-00200">
+            <when>
+                <project>
+                    <artifact artifactId="hibernate-search-analyzers" groupId="org.hibernate" />
+                </project>
+            </when>
+            <perform>
+                <lineitem message="Maven artifact org.hibernate:hibernate-search-analyzers was removed from Hibernate Search 5. We recommend to depend on the appropriate Lucene artifact directly, for example org.apache.lucene:lucene-analyzers-common." />
+            </perform>
+        </rule>
+        <rule id="hsearch-00210">
+            <when>
+                <javaclass references="org.apache.lucene.search.SortField({*})">
+                    <location>CONSTRUCTOR_CALL</location>
```

Comment:
It had to do with type bindings not being found. Fixed by setting the "target" version in the rule test.